### PR TITLE
Consolidate scheduler extension - part 1

### PR DESCRIPTION
**Pull request by mkouba**
- `quarkus-scheduler` contains the API (`io.quarkus.scheduler.Scheduled` and friends), shared logic to collect/validate scheduled methods and a simple scheduler impl (we use `com.cronutils:cron-utils` to parse/validate/migrate CRON expressions, a `ScheduledExecutorService` with one "scheduler" thread that delegates the execution of scheduled methods to the `ExecutorService` obtained from `ExecutorBuildItem`; if `quarkus-scheduler` is present this impl is disabled)
- `quarkus-quartz` depends on `quarkus-scheduler` and reuses most of the build steps and API; however, it's using Quartz as the underlying scheduler impl and provides some quartz-specific configuration
- CRON syntax is configurable through the `quarkus.scheduler.cron-type` property; the default value is `quartz` to retain backward compatibility; other options are `unix` and `cron4j`
- the `Scheduler#startTimer()` method was removed - it was a mistake
- there is a new intergration test because previously scheduler was included in `quarkus-integration-test-main` but there were no assertions

TODO (next PRs):
- update the docs, although almost everything should work as before; mention the `quarkus.scheduler.cron-syntax` property - PR  #5481 opened
- consider adding more quartz-specific configuration (@machi1990 clustered job store etc.) PR opened for clustered jobs https://github.com/quarkusio/quarkus/pull/5529
- maybe add some more tests ;-)

### CODE REVIEW

File: extensions/scheduler/runtime/src/main/java/io/quarkus/scheduler/runtime/SimpleScheduler.java
Comment (dmlloyd) [inline extensions/scheduler/runtime/src/main/java/io/quarkus/scheduler/runtime/SimpleScheduler.java]: One thing I was curious about and I think I mentioned before is, whether it's worth exposing the `ScheduledExecutorService` directly to applications which want to programmatically schedule things.  Maybe as a future enhancement?

**Review by dmlloyd — APPROVED** [review]
Otherwise seems fine to me.

File: extensions/scheduler/runtime/src/main/java/io/quarkus/scheduler/runtime/SimpleScheduler.java
Comment (mkouba) [reply extensions/scheduler/runtime/src/main/java/io/quarkus/scheduler/runtime/SimpleScheduler.java]: Yes, that's why I removed the `Scheduler#startTimer()` method which was basically mimicking `ScheduledExecutorService.schedule()`. In fact, we don't expose any `ExecutorService` ATM and some users coming from the EE world could expect some support of `ManagedExecutorService` and friends.

**Comment by geoand** [comment]
The test failures seem related

**Comment by mkouba** [comment]
@geoand OK, I'll investigate the failure.

**Comment by geoand** [comment]
Cool

**Comment by machi1990** [comment]
Thanks @mkouba for this PR. I'll have a look at it tomorrow and revive my work on quartz clustered jobs.

**Comment by mkouba** [comment]
Ok, we're green now! ;-) Let's wait for @machi1990 's review though...

File: extensions/quartz/deployment/src/main/java/io/quarkus/quartz/deployment/QuartzProcessor.java
Comment (machi1990) [inline extensions/quartz/deployment/src/main/java/io/quarkus/quartz/deployment/QuartzProcessor.java]: Maybe we can produce a `quartz` FeatureBuildItem in this class?

File: extensions/quartz/deployment/src/main/java/io/quarkus/quartz/deployment/QuartzProcessor.java
Comment (machi1990) [inline extensions/quartz/deployment/src/main/java/io/quarkus/quartz/deployment/QuartzProcessor.java]: Should this be 

```suggestion
                "Quartz scheduler 'QuarkusQuartzScheduler'"));
```
? 

likewise below

File: extensions/quartz/runtime/src/main/java/io/quarkus/quartz/runtime/QuartzRuntimeConfig.java
Comment (machi1990) [inline extensions/quartz/runtime/src/main/java/io/quarkus/quartz/runtime/QuartzRuntimeConfig.java]:
```suggestion
     * The size of scheduler thread pool. This will initialize the number of worker threads in the pool.
```

**Review by machi1990 — APPROVED** [review]
Added minor comments, otherwise LGTM. Great progress :-)

I have already started reviving https://github.com/quarkusio/quarkus/pull/3783 based on this work.

File: extensions/quartz/deployment/src/main/java/io/quarkus/quartz/deployment/QuartzProcessor.java
Comment (machi1990) [reply extensions/quartz/deployment/src/main/java/io/quarkus/quartz/deployment/QuartzProcessor.java]: Okay, makes totally sense. Sorry, I had went ahead and added it in my PR . I’ll remove it once this is merged. Thanks.

Comment (mkouba) [reply extensions/quartz/deployment/src/main/java/io/quarkus/quartz/deployment/QuartzProcessor.java]: No problem ;-).

**Comment by machi1990** [comment]
@mkouba CI is green, we can merge this right? I would like to pick up some changes you made in my PR #5529

**Comment by machi1990** [comment]
You rock @mkouba ;-)

**Pull request merged**
